# Test cases automatically generated by Pynguin (https://www.pynguin.eu).
# Please check them before you use them.
import pytest
import test as module_0


# @pytest.mark.xfail(strict=True)
def test_case_0(): # FAILED - TypeError: object of type 'int' has no len()
    str_0 = "1hgO/\x0btJy:"
    int_0 = module_0.length_of_longest_substring(str_0)
    assert int_0 == 10
    int_1 = module_0.length_of_longest_substring(str_0)
    assert int_1 == 10
    solution_0 = module_0.Solution()
    solution_0.lengthOfLongestSubstring(int_0)


def test_case_1(): # PASSED
    str_0 = "L,OP}j<kG^\njt]Vz"
    solution_0 = module_0.Solution()
    int_0 = solution_0.lengthOfLongestSubstring(str_0)
    assert int_0 == 11


# @pytest.mark.xfail(strict=True)
def test_case_2(): # FAILED - TypeError: Solution() takes no arguments
    str_0 = "UY*"
    bytes_0 = b"\xf9\x00\xb4\x84\x1c\x14X\xbc\xb9"
    dict_0 = {str_0: str_0, str_0: str_0, str_0: bytes_0}
    str_1 = "ff&tb06yXO,gom\r"
    int_0 = module_0.length_of_longest_substring(str_1)
    assert int_0 == 14
    module_0.Solution(**dict_0)
